### PR TITLE
fix #15659: bug in decompose(::BigFloat)

### LINKED ISSUE #15659: Bug in comparison of `BigFloat` and rational
The following is reduced from an actual bug in testing my package. Confirmed on both 0.4.5 and current master:

```julia
julia> set_bigfloat_precision(53)
53

julia> big(1/3)   #  this is definitely less than a third
3.3333333333333331e-01

julia> big(1/3) < 1//3
false
```

cc @simonbyrne 

Digging down, this apparently uses the `Base.decompose` function from `hashing2.jl`, so also cc'ing @StefanKarpinski.

### PR DESCRIPTION
**Pull request by StefanKarpinski**
This bug stems from the fact that the bits of a BigFloat are left-aligned in its storage while the bits of a BigInt are right-aligned in its storage. When the precision of a BigFloat is divisible by the BigInt limb size, there is no difference between these, but in other cases, there's a substantial difference. This fix return a `pow` value from `decompose` that accounts for the difference in alignment.

### CODE REVIEW

**Comment by StefanKarpinski** [comment]
Anyone have a clue about that test failure? Why would only the Windows x86_64 test fail?

**Comment by tkelman** [comment]
It's the only platform where C long is not pointer size.

**Comment by StefanKarpinski** [comment]
Ah, that's an excellent hint. Thanks, @tkelman.

**Comment by dpsanders** [comment]
Solves my original issue, many thanks! :+1:

File: base/mpfr.jl [hunk 1]
```diff
@@ -46,7 +46,7 @@ type BigFloat <: AbstractFloat
     prec::Clong
     sign::Cint
     exp::Clong
-    d::Ptr{Culong}
+    d::Ptr{Limb}
```
Comment (tkelman) [inline base/mpfr.jl]: does mpfr use gmp's Limb size for this in C?

Comment (StefanKarpinski) [inline base/mpfr.jl]: It does.

**Pull request merged**